hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakeInfoHelper.java

```diff
@@ -0,0 +1,128 @@
+import com.hedera.node.app.service.token.impl.WritableStakingInfoStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Helper class for mutating staking info in the {@link WritableStakingInfoStore}
+ */
+@Singleton
+public class StakeInfoHelper {
+    private static final Logger log = LogManager.getLogger(StakeInfoHelper.class);
+
+    @Inject
+    public StakeInfoHelper() {}
+
+    /**
+     * Increases the unclaimed stake reward start for the given node by the given amount
+     *
+     * @param nodeId the node's numeric ID
+     * @param amount the amount to increase the unclaimed stake reward start by
+     */
+    public void increaseUnclaimedStakeRewards(
+            @NonNull final Long nodeId, final long amount, @NonNull final WritableStakingInfoStore stakingInfoStore) {
+        requireNonNull(nodeId);
+        requireNonNull(stakingInfoStore);
+
+        final var currentStakingInfo = stakingInfoStore.get(nodeId);
+        final var currentStakeRewardStart = currentStakingInfo.stakeRewardStart();
+        final var newUnclaimedStakeRewardStart = currentStakingInfo.unclaimedStakeRewardStart() + amount;
+
+        final var newStakingInfo = currentStakingInfo.copyBuilder();
+        if (newUnclaimedStakeRewardStart > currentStakeRewardStart) {
+            log.warn(
+                    "Asked to release {} more rewards for node{} (now {}), but only {} was staked",
+                    amount,
+                    nodeId,
+                    newUnclaimedStakeRewardStart,
+                    currentStakeRewardStart);
+            newStakingInfo.unclaimedStakeRewardStart(currentStakeRewardStart);
+        } else {
+            newStakingInfo.unclaimedStakeRewardStart(newUnclaimedStakeRewardStart);
+        }
+
+        stakingInfoStore.put(nodeId, newStakingInfo.build());
+    }
+
+    /**
+     * Awards the stake to the node's stakeToReward or stakeToNotReward depending on the account's decline reward.
+     * If declineReward is true, the stake is awarded to stakeToNotReward, otherwise it is awarded to stakeToReward.
+     * @param nodeId the node's numeric ID
+     * @param account the account stake to be awarded to the node
+     * @param stakingInfoStore the store for the staking info
+     */
+    public void awardStake(
+            @NonNull final Long nodeId,
+            @NonNull final Account account,
+            @NonNull final WritableStakingInfoStore stakingInfoStore) {
+        requireNonNull(nodeId);
+        requireNonNull(account);
+        requireNonNull(stakingInfoStore);
+
+        final var stakeToAward = roundedToHbar(totalStake(account));
+        final var isDeclineReward = account.declineReward();
+
+        final var stakingInfo = stakingInfoStore.get(nodeId);
+        final var copy = stakingInfo.copyBuilder();
+        if (isDeclineReward) {
```

Comment:
Checking we can't overflow `stakeToNotReward` would be a nice safety check. (Seems not in mono-service either tho.)

Reply:
You mean check if > Long.MAX_VALUE ? I have assigned to a long variable, and it should be sufficient right ?